### PR TITLE
Feature/edit target

### PR DESCRIPTION
**Pull request by joaquinTop**
### Feature: *Edit Target*
* Added put request for updating selected target
* Added buttons for delete & update target
* Message displayed when target is successfully updated

### CODE REVIEW

File: src/actions/targetActions.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export function createTargetSuccess(target){
 export const createTarget = (target) => {
   return dispatch => {
     return targetClient.createTarget(target).then(data => {
-      dispatch(createAlert("SideBarContainer", "Target created successfully", "success"));
+      dispatch(createAlert("SideBarContainer", "Target successfully created", "success"));
```
Comment (felipegarcia92) [inline src/actions/targetActions.js]: I looked at the code of the `createAlert` function and I think it's a good idea to refactor those "SideBarContainer" and "success" flags to constants.

I'd declare a goals object with keys, for example:
```
const goals = {
  sidebar: 'SidebarContainer',
  ...etc
}
```
in a constants file, then import it and use it like `constants.sidebar`.

Also, remember that when writing JS, single quotes `'` are preferred.

I suggest you do the same with the alert types, so you don't have to use explicit strings all over the code.

File: src/components/common/Map.js [hunk 2]
```diff
@@ -170,7 +170,11 @@ class Map extends React.Component {
 
     // TARGETS RADIUS
     let circles = this.props.markers.map((venue) => {
-      const color = (this.props.targetSelected === venue) ? 'rgb(48, 188, 247)' : 'rgb(239, 197, 55)';
+
+      const { targetSelected } = this.props;
+      const isSelectedTarget = targetSelected && targetSelected.id === venue.id;
+      const color = isSelectedTarget ? 'rgb(48, 188, 247)' : 'rgb(239, 197, 55)';
```
Comment (felipegarcia92) [inline src/components/common/Map.js]: Declare the color strings as a constants saying more or less what color is it.

File: src/components/targets/TargetForm.js [hunk 3]
```diff
@@ -27,16 +28,21 @@ class TargetForm extends React.Component{
     updateTargetInfo({topic: index});
   }
 
+  onTargetDelete(e){
+    e.preventDefault();
+
```
Comment (felipegarcia92) [inline src/components/targets/TargetForm.js]: Delete the extra new lines here, I don't think they're necessary

File: src/components/targets/TargetForm.js [hunk 4]
```diff
@@ -27,16 +28,21 @@ class TargetForm extends React.Component{
...
-    if (!enabled) {
-      createAlertAction("SideBarContainer", "Max of 10 targets reached", "error");
-      return;
-    }
 
-    if (formMode === "Edit") {
-      deleteTargetAction(currentTarget.id);
+    const { enabled, formMode, currentTarget, createAlertAction, createTargetAction, updateTargetAction } = this.props;
+
+    if (!enabled && formMode === "New") {
```
Comment (felipegarcia92) [inline src/components/targets/TargetForm.js]: Refactor the form modes into constants, just like the other comment above.

File: src/components/home/SideBarContainer.js [hunk 5]
```diff
@@ -91,32 +84,35 @@ export const SideBarContainer = (props) => {
...
-              onClick={logOut}
-              type="button"
-              className="btn btn-danger btn-sign-out">Sign out</button>
           </div>
         </div>
       );
     }
 
-    case "Home":
+    case "Home":{
```
Comment (felipegarcia92) [inline src/components/home/SideBarContainer.js]: Reafactor these content strings into a constant, like the comment above.

**Review by felipegarcia92 — CHANGES_REQUESTED** [review]

File: src/components/targets/TargetForm.js [hunk 6]
```diff
@@ -27,16 +28,21 @@ class TargetForm extends React.Component{
     updateTargetInfo({topic: index});
   }
 
+  onTargetDelete(e){
+    e.preventDefault();
+
+    const { deleteTargetAction, currentTarget } = this.props;
```
Comment (bernabe9) [inline src/components/targets/TargetForm.js]: You are use only the `id` of the `currentTarget`, so you can use a shortcut here:
`const { deleteTargetAction, currentTarget: { id } } = this.props;`

**Review by bernabe9 — CHANGES_REQUESTED** [review]

**Review by felipegarcia92 — APPROVED** [review]
I like this solution and will probably the enum library on future projects

**Review by bernabe9 — APPROVED** [review]

**Pull request merged**